hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsStoreAggregateSource.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.regionserver;
+
+import org.apache.hadoop.hbase.metrics.BaseSource;
+import org.apache.yetus.audience.InterfaceAudience;
+
+
+@InterfaceAudience.Private
+public interface MetricsStoreAggregateSource extends BaseSource {
+  /**
+   * The name of the metrics
+   */
+  String METRICS_NAME = "Stores";
+
+  /**
+   * The name of the metrics context that metrics will be under.
+   */
+  String METRICS_CONTEXT = "regionserver";
+
+  /**
+   * Description
+   */
+  String METRICS_DESCRIPTION = "Metrics about Stores under a region";
+
+  /**
+   * The name of the metrics context that metrics will be under in jmx
+   */
+  String METRICS_JMX_CONTEXT = "RegionServer,sub=" + METRICS_NAME;
```

Comment:
This model doesn't seem right. There is no such thing as a Store on a RegionServer. The RegionServer hosts Regions. A Region hosts Stores.  This is an aggregate of all Stores on the RegionServer?

I could imagine RegionServer,sub=Region,sub=Store... with a Bean per Store but then we'd probably have too many MBeans. 

Do we have a Region at this level in MBean hierarchy?

Reply:
I was just trying to push all the store level metric per region under this. that is why collected it under RegionServer.

Reply:
And seeing ur other comment -  this is where yo uwill get per region under each store I have how many requests. If I want to remove this then I may have to do something like how it has been done for Table vs Store - we will have region vs store (and it will be at the region level only).

Reply:
I think to avoid this problem of having another metric exposed at Store level which will list all the region names and its corresponding read metrics and also we have a region level metric where we expose across all stores in that region how many reads happened.  
Both come under different Mbean - one is sub=Regions and another is sub=Stores. (this is added newly in this patch)

The patch already has an aggregated metric at the Table level where we report a metric 

> NameSpace_<Namespacename>table<tablename>_store<storename>_metric<metricName>
We will just use this and add a similar metric at the region level which aggregates it per store
as we have for table level. Will that be ok ? 
@saintstack - what you think?